beacon-chain/rpc/prysm/v1alpha1/validator/server.go

```diff
@@ -123,6 +123,9 @@ func (vs *Server) ValidatorIndex(ctx context.Context, req *ethpb.ValidatorIndexR
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not determine head state: %v", err)
 	}
+	if st == nil {
```

Comment:
```suggestion
	if st == nil || st.IsNil() {
```
this is more bulletproof